packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -366,12 +373,24 @@ class NavigatorState extends State<Navigator> {
       route.didChangeNext(null);
       if (oldRoute != null)
         oldRoute.didChangeNext(route);
-      config.observer?.didPush(route, oldRoute);
+      _didPush(route, oldRoute);
     });
     assert(() { _debugLocked = false; return true; });
     _cancelActivePointers();
   }
 
+  void _didPush(Route<dynamic> route, Route<dynamic> previousRoute) {
+    config.observer?.didPush(route, previousRoute);
+    assert(() {
+      String name = route?.settings?.name;
+      developer.postEvent('Flutter.RouteChanged', <String, dynamic>{
+        'route': name,
+        'changeType': 'push'
+      });
+      return true;
+    });
```

Comment:
I don't think this code makes sense here.  It should be in PageRoute or one of the other subclasses that already has a notion of having a name.